virtcontainers/agent.go

```diff
@@ -144,6 +144,9 @@ type agent interface {
 	// supported by the agent.
 	capabilities() capabilities
 
+	// disconnect will disconnect the connection to the agent
+	disconnect()
```

Comment:
I don't understand why this cannot be:

```
disconnect() error
```

Reply:
I was thinking that we cannot do more recovery if disconnect() ever fails. I can add it back if you want though.

Reply:
@bergwolf Will be useful to return the error to atleast log it even if it we cannot recover.